apps/service_define/lib/define/view_state.ex

```diff
@@ -0,0 +1,41 @@
+defmodule Define.ViewState do
+  use GenServer
+
+  def event(pid, type, payload) do
+    GenServer.call(pid, {:event, type, payload})
+  end
+
+  def state(pid) do
+    GenServer.call(pid, :state)
+  end
+
+  @impl true
+  def init(state) do
+    {:ok, state}
+  end
+
+  def start_link(init_opts) do
+    server_opts = Keyword.take(init_opts, [:name])
+    GenServer.start_link(__MODULE__, default_state(), server_opts)
+  end
+
+  @impl true
+  def handle_call({:event, type, payload}, _from, state) do
+    new_state = update_state(state, type, payload)
+    {:reply, new_state, new_state}
+  end
+
+  @impl true
+  def handle_call(:state, _from, state) do
+    {:reply, state, state}
+  end
+
+  @spec default_state :: map()
+  def default_state() do
+    %{"greeting" => "Hola Mundo!"}
```

Comment:
I take it that this is just part of proving out the thing works?

Reply:
Yup, thats the idea